src/controllers/user.controller.ts

```diff
@@ -57,4 +58,42 @@ const checkReferralExist = async (req: Request, res: Response) => {
     }
 };
 
-export { getUser, getAllUsers, checkReferralExist };
+const getUserWithAllInfo = async (req: Request, res: Response) => {
+    try {
+        const id: number = Number(req.params.id);
+        const user: User | null = await getUserByIdWithAllInfo(id);
+        let userDTO = null;
+
+        if (user instanceof User) {
+            userDTO = instanceToPlain(user, { groups: ['profil'], excludeExtraneousValues: true });
+            res.status(200).json(userDTO);
+        } else {
+            res.status(400).json(`This user doesn't exist`);
+        }
+    } catch (e) {
+        logger.error(`getUser error`, e);
+        res.status(500).json({ message: 'Oops !, an error has occurred.' });
+    }
+};
+
+const updateAllInfo = async (req: Request, res: Response) => {
+    try {
+        const id: number = Number(req.params.id);
+        const body = req.body;
+        let infoUpdate: number | boolean | null = null;
+
+        if(body.username) infoUpdate = await setUser(id, body);
```

Comment:
J'approuve pas cette façon de faire. Tu fais juste un controle sur le username alors que tu sais qu'il est déjà rempli, vu qu'il est obligatoire.
Pour faire un set user (alors que c'est une mise à jour) sans vérifier les données complétées.

Reply:
C'est vrai que j'aurais dû contrôler les données, je m'en occupe. Par contre, la condition est là pour vérifier quelle table on doit modifier, d'où l'utilisation du nom d'utilisateur car, comme tu l'as dit, je sais qu'il sera rempli. Donc, s'il se trouve dans le body, c'est qu'on veut modifier la table User.

Reply:
C'est encore pire du coup.
Une route, une table
UpdateUser => modifie le user
UpdateSocial => modifie le social media

Reply:
Justement, je ne voulais pas faire cela de cette manière car la demande se fait par rapport à un utilisateur. Même si on a 3 tables, si l'on regarde, 2 des tables ne seraient pas possibles sans la création d'un utilisateur. C'est donc la raison pour laquelle j'ai préféré choisir cette façon, cela évite d'avoir des routes en plus et, vu que ça concerne uniquement l'utilisateur. 

C'était ma première idée mais je trouvais qu'on dupliqué du code, on risque quoi avec ma ma façon de faire ?

Reply:
C'est noté et merci pour l'article 😃